src/status_printer.h

```diff
@@ -36,9 +36,11 @@ struct StatusPrinter : Status {
   void BuildStarted() override;
   void BuildFinished() override;
 
-  void Info(const char* msg, ...) override;
-  void Warning(const char* msg, ...) override;
-  void Error(const char* msg, ...) override;
+  void Info(const char* msg, ...) const override;
+  void Warning(const char* msg, ...) const override;
+  void Error(const char* msg, ...) const override;
+
+  bool IsValidProgressStatus(const char* progress_status_format) const;
```

Comment:
nit: Please make this method static.

Reply:
Also, document it properly and make it return an error string, instead of calling Warning(), i.e. let the caller decide what to do, e.g.:

```
/// Validate the status format string passed as an argument to verify it only contains
/// valid escape sequences. On success return true, on failure, return false and sets \arg error.
static bool IsValidProgressStatusFormat(const char* status_format, std::string* err);
```